airframe-http/src/main/scala/wvlet/airframe/http/HttpMessage.scala

```diff
@@ -114,11 +116,12 @@ trait HttpMessage[Raw] {
   def withContentTypeMsgPack: Raw                       = withContentType(HttpHeader.MediaType.ApplicationMsgPack)
   def withContentLength(length: Long): Raw              = withHeader(HttpHeader.ContentLength, length.toString)
   def withDate(date: String): Raw                       = withHeader(HttpHeader.Date, date)
+  def withDate(date: Date): Raw                         = withHeader(HttpHeader.Date, FinagleMessage.httpDateFormat(date))
```

Comment:
The implementation for `Message. httpDateFormat` is so small that we should implement this feature without `finagle-base-http`.

https://github.com/twitter/finagle/blob/34167989a6dc964e24e20212516fba89fa58bb23/finagle-base-http/src/main/scala/com/twitter/finagle/http/Message.scala#L642-L658

Reply:
ZonedDateTime is not supported in Scala.js, so we need to use Instant, which is implemented in https://github.com/scala-js/scala-js-java-time
. 

I'm not sure about java.util.Date. I guess this is not supported too. 

It's better to make it an extension only for JVM project, using some implicit class

Reply:
For example, Date time is not supported in Scala.js, so airframe-codec supports Date and ZonedDateTime only for JVM projects https://github.com/wvlet/airframe/blob/master/airframe-codec/.jvm/src/main/scala/wvlet/airframe/codec/JavaTimeCodec.scala

Reply:
Two options:
- Use withDate(instant) 
- Add an implicit class RichMessage(val message:HttpMessage) extends AnyVal { def withDate(...) } only for airframe-http (JVM)

I think the first one is easier

Reply:
The third option: implement Scala JVM/Scala.js specific httpDateFormatter using Compat.scala files